### PR TITLE
[WIP] Decouple from suit and prepare for jet

### PR DESCRIPTION
**Pull request by AidanCurrah**
Relies on [PR for CMS](https://github.com/onespacemedia/cms/pull/153). Until both a ready do not merge.

This decouples the PT from suit and gets it ready for the migration to Jet. It uses django-adminsortable for the sortables instead and removes most of our admin template overrides.

Bugs
---

* [x] Section field hiding/showing does not work when first adding a section - it shows all fields
* [x] WYSIWYG cannot be focused when first adding a section
* [x] Update login page to not have `user and password: admin`
* [x] Sitemap module needs some styling (see below)
* [x] Sitemap module doesn't have ordering controls
* [x] Sitemap module is not draggable on the admin homepage
* [x] Disable save button on click as you can click multiple times to add a few pages at once
* [x] 404 on admin homepage: `(index):59 GET http://localhost:3000/static/admin/js/jquery.js 404 (Not Found)`
* [x] Exception on the 'Change file' page (traceback below)
* [x] Image editor is not loading on the change file page
* [x] Rearrange fieldsets on `ModelAdmin`s on the default apps so that the most important information (e.g. text on news articles) is shown on the first tab
* [x] Clicking the "+" next to the widget for an `ImageRefField` doesn't work - try adding a new image for a news article, for example
* [x] On that note, the edit link there doesn't work either (greyed out) 
* [x] Closing the edit screen modal (brought up by clicking 'Edit' on the front-end editing bar) raises an exception (so the modal never closes)
* [x] Cannot insert new images into the media library and insert them into a WYSIWYG at the same time (traceback in the comments)

Enhancements
---

* [x] Can we remove the delay when showing submenus in the admin?
* [x] Can we remove 'Application page' from submenus? (yep - with a custom menu and not supplying `app_label`)
* [x] It'd be neat if django-sortedm2m works with Jet!
* [x] Improve 'add page' screen

### CODE REVIEW

**Comment by lewiscollard** [comment]
Updating original comment to add bug with sections JS (hiding/showing fields doesn't work when first adding a section).

Edit: Seems that the class names aren't what's expected. Existing items have the class `dynamic-contentsection_set` which matches the [selector `[class*=section_set]`](https://github.com/onespacemedia/project-template/blob/9e3e8465816c3849f16f243460f08d0808fb3055/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.package_name%7D%7D/apps/sections/templates/admin/pages/page/sections.js#L9) but newly-added ones do not.

**Comment by lewiscollard** [comment]
Added two issues with the sitemap module to the checklist.

screen shot 2018-08-15 at 12 00 52

**Comment by lewiscollard** [comment]
New item on checklist: Exception when saving the page when using the edit bar on the front end:

```
bundle.min.js:1 TypeError: Cannot read property 'find' of undefined
    at s.processPopupResponse (bundle.min.js:1)
    at bundle.min.js:1
    at s.closePopup (bundle.min.js:1)
    at HTMLDivElement.<anonymous> (bundle.min.js:1)
    at Function.each (bundle.min.js:8)
    at ot.fn.init.each (bundle.min.js:7)
    at s.findPopupResponse (bundle.min.js:1)
    at s.run (bundle.min.js:1)
    at HTMLDocument.<anonymous> (bundle.min.js:1)
    at u (bundle.min.js:8) "TypeError: Cannot read property 'find' of undefined
    at s.processPopupResponse (http://localhost:3000/static/jet/js/build/bundle.min.js:1:21701)
    at http://localhost:3000/static/jet/js/build/bundle.min.js:1:21195
    at s.closePopup (http://localhost:3000/static/jet/js/build/bundle.min.js:1:21432)
    at HTMLDivElement.<anonymous> (http://localhost:3000/static/jet/js/build/bundle.min.js:1:21595)
    at Function.each (http://localhost:3000/static/jet/js/build/bundle.min.js:8:1086)
    at ot.fn.init.each (http://localhost:3000/static/jet/js/build/bundle.min.js:7:30940)
    at s.findPopupResponse (http://localhost:3000/static/jet/js/build/bundle.min.js:1:21536)
    at s.run (http://localhost:3000/static/jet/js/build/bundle.min.js:1:22998)
    at HTMLDocument.<anonymous> (http://localhost:3000/static/jet/js/build/bundle.min.js:1:23182)
    at u (http://localhost:3000/static/jet/js/build/bundle.min.js:8:24846)"
run @ bundle.min.js:1
(anonymous) @ bundle.min.js:1
u @ bundle.min.js:8
fireWith @ bundle.min.js:8
ready @ bundle.min.js:8
a @ bundle.min.js:7
```

**Comment by lewiscollard** [comment]
Added to checklist: Problem inserting images from the library into WYSIWYGs.

```
bundle.min.js?v=1.0.7:1 Uncaught TypeError: Cannot read property 'is' of undefined
    at s.processPopupResponse (bundle.min.js?v=1.0.7:1)
    at bundle.min.js?v=1.0.7:1
    at s.closePopup (bundle.min.js?v=1.0.7:1)
    at HTMLAnchorElement.<anonymous> (bundle.min.js?v=1.0.7:1)
    at HTMLAnchorElement.dispatch (bundle.min.js?v=1.0.7:9)
    at HTMLAnchorElement.m.handle (bundle.min.js?v=1.0.7:9)
```

**Comment by lewiscollard** [comment]
Edited: 404 on admin homepage, split out some "would be nice".

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/templates/admin/auth/user/change_list.html [hunk 1]
```diff
@@ -1 +1,6 @@
 {% extends "admin/change_list.html" %}
+
+{# TODO Discuss this change in the CMS instead of overriding it in a template. #}
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/templates/admin/auth/user/change_list.html]: Does this template do anything at all?

**Comment by lewiscollard** [comment]
<details>
<summary>Issue fixed: modernise the 'Add page' screen</summary>
Added: I think the 'add page' screen could look a bit more in place. Devsign time!

screen shot 2018-08-15 at 12 52 09

Long-standing problems with the list only being able to handle one line of text (convert this to use flexbox or grid):

screen shot 2018-08-15 at 12 54 59

More of a CMS issue as these are being overridden in the project template, but the default 'content' and 'link' icons look even more out of place in Jet (let's get the PT overrides into the CMS):
screen shot 2018-08-15 at 12 56 45

screen shot 2018-08-15 at 12 57 07
</details>

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/templates/admin/auth/user/change_list.html (hunk 1 above)
Comment (AidanCurrah) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/templates/admin/auth/user/change_list.html]: it's overwriting the CMS template this is:
```
{% extends "admin/change_list.html" %}


{% block object-tools-items %}
    <a class="btn" href="{% url 'admin:auth_user_invite' %}">Invite user</a>
    {{block.super}}
{% endblock %}
```
As the link was throwing errors and I didn't know if anyone ever used the user inviting feature.

Comment (lewiscollard) [reply {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/templates/admin/auth/user/change_list.html]: Delete it from both CMS and PT. If we want to add back user inviting we can bring it back at a project template level.

**Comment by AidanCurrah** [comment]
Relevant Issue for sorted-m2m: https://github.com/gregmuellegger/django-sortedm2m/issues/73

**Comment by lewiscollard** [comment]
New bug added: Popup for adding an image on an `ImageRefField` doesn't seem to work. e.g. for a news article, next to the 'Image' field, click the 'Add' button - it takes you away from the page, and saving the image gives you a blank page.

**Comment by lewiscollard** [comment]
<details>
<summary>Issue fixed: exception on image file change form</summary>


Added two bugs on the media file page. The exception:

```
change_form.js:5 Uncaught TypeError: $ is not a function
    at change_form.js:5
    at change_form.js:20
(anonymous) @ change_form.js:5
(anonymous) @ change_form.js:20
15:22:16.533 prepopulate_init.js:3 Uncaught TypeError: $ is not a function
    at prepopulate_init.js:3
    at prepopulate_init.js:10
```
</details>

**Comment by lewiscollard** [comment]
<details>
<summary>Issue fixed: exception on media file change form</summary>

Also, this exception happens when you go to a non-image media file page:

```
change_form.js:5 Uncaught TypeError: $ is not a function
    at change_form.js:5
    at change_form.js:20
(anonymous) @ change_form.js:5
(anonymous) @ change_form.js:20
15:25:10.650 prepopulate_init.js:3 Uncaught TypeError: $ is not a function
    at prepopulate_init.js:3
    at prepopulate_init.js:10
```
</details>

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/dashboard.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from jet.dashboard import modules
+from osm_jet.dashboard import Dashboard
+
+
+class OSMDashboard(Dashboard):
+    def init_with_context(self, context):
+        '''
+            Add dashboard widgets for the site here.
+
+            Any widgets you add under `self.avaliable_children` will be avaliable in the widget selector.
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/dashboard.py]: available* both times

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/models.py [hunk 3]
```diff
@@ -1,10 +1,116 @@
...
+from django.contrib.auth.models import (AbstractBaseUser, BaseUserManager,
+                                        PermissionsMixin)
 from django.db import models
 from django.utils.functional import cached_property
+from django.utils.timezone import now
 
 from ...utils.models import HasLinkMixin
 
 
+class UserManager(BaseUserManager):
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/models.py]: I would really like this to be broken out into a separate PR, and definitely not in the `site` app.

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/dashboard.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            Any widgets you add under `self.children` will be added to a user's dashboard by default.
+        '''
+
+        self.available_children.append(modules.AppList)
+        self.available_children.append(modules.ModelList)
+        self.available_children.append(modules.RecentActions)
+        self.available_children.append(modules.LinkList)
+        self.children.append(modules.RecentActions(
+            'Recent Actions',
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/site/dashboard.py]: Use sentence case - "Recent actions".

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/sections/admin.py [hunk 5]
```diff
@@ -1,22 +1,27 @@
...
     model = ContentSection
     extra = 0
     filter_horizontal = ['people']
     fk_name = 'page'
 
     class Media:
-        js = [reverse_lazy('admin_sections_js')]
+        js = [
+            reverse_lazy('admin_sections_js')
+        ] + JetCompactInline.Media.js
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/sections/admin.py:19]: Would it not be better for `Media` to inherit from `JetCompactInline.Media`?

**Review by lewiscollard — CHANGES_REQUESTED** [review]

File: {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/careers/admin.py [hunk 6]
```diff
@@ -68,7 +67,7 @@ def get_form(self, request, obj=None, **kwargs):
 
 
 @admin.register(CareerLocation)
-class LocationAdmin(VersionAdmin):
+class LocationAdmin(admin.ModelAdmin):
```
Comment (lewiscollard) [inline {{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/apps/careers/admin.py]: This should continue to inherit from VersionAdmin.

**Review by lewiscollard — CHANGES_REQUESTED** [review]

**Review by lewiscollard — APPROVED** [review]
